src/main/java/nextstep/subway/domain/policy/AddSectionPolicyFactory.java

```diff
@@ -0,0 +1,28 @@
+package nextstep.subway.domain.policy;
+
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.Sections;
+import nextstep.subway.domain.Station;
+
+import java.util.List;
+
+public class AddSectionPolicyFactory {
+
+    public static AddSectionPolicy of(Sections sections, List<Section> sectionList, Section newSection, Station firstStation, Station lastStation) {
+        if (sections.isEmpty()) {
+            return new AddEmptySectionPolicy(sections, sectionList, newSection);
+        }
+
+        if (newSection.getDownStation().equals(firstStation)) {
+            return new AddFirstSectionPolicy(sections, sectionList, newSection);
+        }
+
+        if (newSection.getUpStation().equals(lastStation)) {
+            return new AddLastSectionPolicy(sections, sectionList, newSection);
+        }
+
+        return new AddBetweenSectionPolicy(sections, sectionList, newSection);
```

Comment:
분기문의 위치만 달라졌을뿐, 여기도 동일한 이유로 리팩토링 대상입니다. 

```suggestion
        return addSectionPolicies.stream().filter(AddSectionPolicy::supported).findAny().orElseThrow(XXXException::new);
```

이런식으로 관리주체가 직접하나하나 분기를 통해 적절한 정책을 찾는게 아니라 정책 자신이 해당 조건을 만족하는지에 대해 반환하도록 하는게 좋을 것 같아요.